### PR TITLE
improve errors types to enable auto-complete

### PR DESCRIPTION
**Pull request by kotarella1110**
Improved `type` and `types` field in errors type.
As shown below, auto-complete is enabled and DX is improved!!

スクリーンショット 2020-08-18 12 22 28
スクリーンショット 2020-08-18 12 20 14
スクリーンショット 2020-08-18 12 40 54

Of course, as before, we can use keys other than completion suggestions.

スクリーンショット 2020-08-18 12 24 47
スクリーンショット 2020-08-18 12 41 08

### CODE REVIEW

File: src/types/utils.ts [hunk 1]
```diff
@@ -14,6 +14,10 @@ export type LiteralToPrimitive<T extends any> = T extends string
   ? boolean
   : T;
 
+export type LiteralUnion<T extends U, U extends Primitive> =
+  | T
+  | (U & { _?: never });
```
Comment (bluebill1049) [inline src/types/utils.ts:19]: could you explain what this does?

**Review by bluebill1049 — APPROVED** [review]
Really impressive stuff for better DX.

File: src/types/utils.ts (hunk 1 above)
Comment (kotarella1110) [reply src/types/utils.ts:19]: @bluebill1049 please see here: https://github.com/microsoft/TypeScript/issues/29729

Comment (kotarella1110) [reply src/types/utils.ts:19]: string literal inherits string, but it does not inherit `string & {}`, so string literal and `string & {}` are treated as distinguishable types.

Comment (bluebill1049) [reply src/types/utils.ts:19]: nice! thanks for the info @kotarella1110

**Pull request merged**